### ISSUE TITLE
random_resize in yolox_base.py makes performance degradation for rectangular input

### ISSUE THREAD

**Issue by GodOfSmallThings** [Megvii-BaseDetection/YOLOX#667]
```
    def random_resize(self, data_loader, epoch, rank, is_distributed):
        tensor = torch.LongTensor(2).cuda()

        if rank == 0:
            size_factor = self.input_size[1] * 1.0 / self.input_size[0]
            if not hasattr(self, 'random_size'):
                min_size = int(self.input_size[0] / 32) - self.multiscale_range
                max_size = int(self.input_size[0] / 32) + self.multiscale_range
                self.random_size = (min_size, max_size)
            size = random.randint(*self.random_size)
            size = (int(32 * size), 32 * int(size * size_factor))
            tensor[0] = size[0]
            tensor[1] = size[1]
```

When size_factor is not integer (e.g. 2 or 1), the training result is not accurate. Bboxes are shifted from the object or are not detected. For example, input size of (416, 768) and using random_resize makes big performance degradation.

**Comment by ruinmessi** [comment]
Have you tested with the newest version? We fix a similar bug in #620

**Comment by GodOfSmallThings** [comment]
@ruinmessi 
Thank you, it helped.
